src/web/lib/components/ThemeColorsEditor/index.js

```diff
@@ -2,28 +2,65 @@ import React from "react";
 import classnames from "classnames";
 import { SketchPicker } from "react-color";
 import onClickOutside from "react-onclickoutside";
-import { colorLabels, colorsWithAlpha } from "../../../../lib/constants";
+import { colorLabels, colorsWithAlpha, ESC } from "../../../../lib/constants";
 import { colorToCSS } from "../../../../lib/themes";
 import Metrics from "../../../../lib/metrics";
 
 import "./index.scss";
 
 class ThemeColorsEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: false,
+      colorName: null
+    };
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+  }
+
+  handleClick(name) {
+    const { setSelectedColor } = this.props;
+    setSelectedColor({ name });
+    if (this.state.selected && this.state.colorName !== name) {
+      this.setState({selected: this.state.selected, colorName: name });
+    } else {
+      this.setState({ selected: !this.state.selected, colorName: name });
+    }
+  }
+
   handleClickOutside() {
     const { selectedColor, setSelectedColor } = this.props;
     if (selectedColor !== null) {
       setSelectedColor({ name: null });
     }
+    this.setState({ selected: false, colorName: null });
+  }
+
+  handleKeyPress(event) {
+    if (event.keyCode === ESC) {
+      this.setState({ selected: false, colorName: null });
```

Comment:
I think `setSelectedColor({ name: null })` on ESC would be simpler here - basically copying what `handleClickOutside` does without local state